### PR TITLE
Update elvish to commit fda62502ad7f (fixes nodaemon)

### PR DESCRIPTION
**Pull request by MDr164**
Signed-off-by: Marvin Drees <marvin.drees@9elements.com>

### CODE REVIEW

**Review by hugelgupf — DISMISSED** [review]

**Comment by hugelgupf** [comment]
tests are failing tho

**Comment by rminnich** [comment]
hmm.
=== RUN   TestGBBBuild
    log.go:19: Gobusybox error: non-module go build failed: error building go package in "/tmp/TestGBBBuild4214605674/001/src/bb.u-root.com/bb": ../../github.com/u-root/u-root/vendor/src.elv.sh/pkg/edit/editor.go:98:12: pattern init.elv: no matching files found
        ../../github.com/u-root/u-root/vendor/src.elv.sh/pkg/mods/epm/epm.go:6:12: pattern epm.elv: no matching files found
        ../../github.com/u-root/u-root/vendor/src.elv.sh/pkg/mods/readlinebinding/readlinebinding.go:6:12: pattern readline-binding.elv: no matching files found
        , exit status 1
    gbb_test.go:32: Build(&{map[] map[]}, {GOARCH=amd64 GOOS=linux GOROOT=/usr/local/go GOPATH=/go CGO_ENABLED=0 0xc00092e060 [../test/foo ../../../cmds/core/elvish] /tmp/TestGBBBuild4214605674/001 bbin}); preserving bb generated source directory at /tmp/TestGBBBuild4214605674/001 due to error. To reproduce build, `cd /tmp/TestGBBBuild4214605674/001/src/bb.u-root.com/bb` and `GO111MODULE=off GOPATH=/tmp/TestGBBBuild4214605674/001 go build` != nil
--- FAIL: TestGBBBuild (2.38s)

we're not missing some files are we?

**Comment by MDr164** [comment]
> we're not missing some files are we?

Well those files have been renamed. From `elv_init.go` to `init.elv` for example. I guess gbb is therefore ignoring them as they aren't .go files.

**Comment by MDr164** [comment]
> > we're not missing some files are we?
> 
> Well those files have been renamed. From `elv_init.go` to `init.elv` for example. I guess gbb is therefore ignoring them as they aren't .go files.

Update on that. Everything builds fine and the test also works locally just fine. Just CircleCI does _something_ differently and therefore it fails there.

File: pkg/uroot/builder/gbb_test.go [hunk 1]
```diff
@@ -9,11 +9,13 @@ import (
 
 	gbbgolang "github.com/u-root/gobusybox/src/pkg/golang"
 	"github.com/u-root/u-root/pkg/golang"
+	"github.com/u-root/u-root/pkg/testutil"
 	"github.com/u-root/u-root/pkg/ulog/ulogtest"
 	"github.com/u-root/u-root/pkg/uroot/initramfs"
 )
 
 func TestGBBBuild(t *testing.T) {
+	testutil.SkipIfNotRoot(t) // Why is this needed on CircleCI?
```
Comment (hugelgupf) [inline pkg/uroot/builder/gbb_test.go:18]: Yo this just skips the test altogether in all environments, doesn't it? Don't do this?

**Review by hugelgupf — CHANGES_REQUESTED** [review]

File: pkg/uroot/builder/gbb_test.go (hunk 1 above)
Comment (hugelgupf) [reply pkg/uroot/builder/gbb_test.go:18]: The unit test runs as not-root, so it's skipped. And when run in a VM, there's no Go compiler, so it's also skipped.

Comment (MDr164) [reply pkg/uroot/builder/gbb_test.go:18]: But in gotest_test.go those tests are invoked again? And this is also a mechanism used to get coverage from some VM tests no? Also I wouldn't have to use it if you can tell me why it only specifically fails in CircleCI and nowhere else. I would rather do a `testutil.SkipIfInCircleCI(t)` if it existed.

Comment (MDr164) [reply pkg/uroot/builder/gbb_test.go]: Update: GO111MODULE=off broke it as those additional files were not correctly recognized by gbb and therefore not copied over to the temporary source tree. Omitting that variable and using Golangs default (e.g. on) has no issues with it.

**Review by hugelgupf — APPROVED** [review]

**Pull request merged**